### PR TITLE
Convert controller cronjob files to structured logs

### PR DESCRIPTION
**Pull request by scottilee**
#### What type of PR is this?
/kind cleanup
/sig instrumentation

#### What this PR does / why we need it:
Converts log statements to use structured logging.

#### Which issue(s) this PR fixes:
N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/1602-structured-logging/kep.yaml
```

### CODE REVIEW

File: pkg/controller/cronjob/cronjob_controller.go [hunk 1]
```diff
@@ -184,7 +184,7 @@ func cleanupFinishedJobs(cj *batchv1.CronJob, js []batchv1.Job, jc jobControlInt
 	// Update the CronJob, in case jobs were removed from the list.
 	if _, err := cjc.UpdateStatus(cj); err != nil {
 		nameForLog := fmt.Sprintf("%s/%s", cj.Namespace, cj.Name)
```
Comment (pacoxu) [inline pkg/controller/cronjob/cronjob_controller.go:186]: `nameForLog := fmt.Sprintf("%s/%s", cj.Namespace, cj.Name)`

Per https://github.com/kubernetes/klog/blob/master/klog.go#L1583-L1595, klog.KObj  is used on Pod(namespace and name)

Not sure if `nameForLog` is a better way.

**Comment by pacoxu** [comment]
Need squash and follow https://github.com/kubernetes/kubernetes/pull/100817#issuecomment-815358999 to rebase.

File: pkg/controller/cronjob/cronjob_controller.go (hunk 1 above)
Comment (scottilee) [reply pkg/controller/cronjob/cronjob_controller.go:186]: Will fix this.

**Comment by scottilee** [comment]
@pacoxu I've been running `git pull upstream/master` instead of `git fetch upstream` and `git rebase upstream/master` so it's making squashing difficult, any suggestions?

**Comment by pacoxu** [comment]
Refer to https://github.com/kubernetes/community/blob/master/contributors/guide/github-workflow.md#squash-commits ?

**Comment by scottilee** [comment]
@pacoxu Fixed the rebase issue, let me know if there's anything else I need to address.

File: pkg/controller/cronjob/cronjob_controller.go [hunk 2]
```diff
@@ -195,12 +194,11 @@ func removeOldestJobs(cj *batchv1.CronJob, js []batchv1.Job, jc jobControlInterf
...
 	}
 
-	nameForLog := fmt.Sprintf("%s/%s", cj.Namespace, cj.Name)
-	klog.V(4).Infof("Cleaning up %d/%d jobs from %s", numToDelete, len(js), nameForLog)
+	klog.V(4).InfoS("Cleaning up jobs", "numToDelete", numToDelete, "jobCount", len(js), "cronjob", klog.KObj(cj))
 
 	sort.Sort(byJobStartTime(js))
 	for i := 0; i < numToDelete; i++ {
-		klog.V(4).Infof("Removing job %s from %s", js[i].Name, nameForLog)
+		klog.V(4).InfoS("Removing job", "cronjobName", js[i].Name, "cronjob", klog.KObj(cj))
```
Comment (yangjunmyfm192085) [inline pkg/controller/cronjob/cronjob_controller.go:201]: I have a question，if cronjob contains cronjobName? if so, let's only keep cronjob,if not, need modify the name?

Comment (scottilee) [reply pkg/controller/cronjob/cronjob_controller.go]: It's actually a batchjob so I updated it. Let me know if there's something else you prefer.

**Comment by yangjunmyfm192085** [comment]
I think this is part duplicated with  https://github.com/kubernetes/kubernetes/pull/99279

**Comment by scottilee** [comment]
Rebased.

**Review by yangjunmyfm192085 — COMMENTED** [review]
/lgtm

**Pull request closed without merging**